### ISSUE TITLE
Training for yolo11m-seg at the process of training , the metrics is normal. But  which is abnormal at the process of validation.

### ISSUE THREAD

**Issue by cjdfree** [ultralytics/ultralytics#22311]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Val

### Bug

Training for yolo11m-seg at the process of training , the metrics is normal. But  which is abnormal at the process of validation.

### Environment

(D:\Program Files\anaconda\environments\ultralytics-improve) PS E:\pythonProject\ComputerVisionProject\YOLO\ultralytics-segment-GAN> yolo check
Ultralytics 8.3.206  Python-3.10.16 torch-2.7.0+cu118 CUDA:0 (NVIDIA GeForce RTX 3050, 8192MiB)
Setup complete  (16 CPUs, 31.8 GB RAM, 1322.1/1678.0 GB disk)

OS                     Windows-10-10.0.22000-SP0
Environment            Windows
Python                 3.10.16
Install                pip
Path                   D:\Program Files\anaconda\environments\ultralytics-improve\Lib\site-packages\ultralytics
RAM                    31.84 GB
Disk                   1322.1/1678.0 GB
CPU                    13th Gen Intel Core(TM) i5-13400F
CPU count              16
GPU                    NVIDIA GeForce RTX 3050, 8192MiB
GPU count              1
CUDA                   11.8

numpy                   2.2.6>=1.23.0
matplotlib              3.10.3>=3.3.0
opencv-python           4.11.0.86>=4.6.0
pillow                  11.2.1>=7.1.2
pyyaml                  6.0.2>=5.3.1
requests                2.28.2>=2.23.0
scipy                   1.15.3>=1.4.1
torch                   2.7.0+cu118>=1.8.0
torch                   2.7.0+cu118!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision             0.22.0+cu118>=0.9.0
psutil                  7.0.0
polars                  1.33.1
ultralytics-thop        2.0.14>=2.0.0

### Minimal Reproducible Example

traning config:
`task: segment
mode: train
model: runs\segment\mixed\train-mixed_6_1\weights\last.pt
data: E:\pythonProject\ComputerVisionProject\YOLO\ultralytics-segment-GAN\my_project\datasets\mixed_crack_dataset\6_1\mixed_6_1.yaml
epochs: 800
time: null
patience: 50
batch: 4
imgsz: 1024
save: true
save_period: -1
cache: false
device: '0'
workers: 8
project: ./runs/segment/mixed
name: train-mixed_6_1
exist_ok: false
pretrained: false
optimizer: AdamW
verbose: true
seed: 0
deterministic: true
single_cls: false
rect: false
cos_lr: false
close_mosaic: 10
resume: runs\segment\mixed\train-mixed_6_1\weights\last.pt
amp: false
fraction: 1.0
profile: false
freeze: null
multi_scale: false
compile: false
overlap_mask: true
mask_ratio: 4
dropout: 0.0
val: true
split: val
save_json: false
conf: null
iou: 0.7
max_det: 300
half: false
dnn: false
plots: true
source: null
vid_stride: 1
stream_buffer: false
visualize: false
augment: false
agnostic_nms: false
classes: null
retina_masks: false
embed: null
show: false
save_frames: false
save_txt: false
save_conf: false
save_crop: false
show_labels: true
show_conf: true
show_boxes: true
line_width: null
format: torchscript
keras: false
optimize: false
int8: false
dynamic: false
simplify: true
opset: null
workspace: null
nms: false
lr0: 0.01
lrf: 0.01
momentum: 0.937
weight_decay: 0.0005
warmup_epochs: 3.0
warmup_momentum: 0.8
warmup_bias_lr: 0.1
box: 7.5
cls: 0.5
dfl: 1.5
pose: 12.0
kobj: 1.0
nbs: 64
hsv_h: 0.015
hsv_s: 0.7
hsv_v: 0.4
degrees: 0.0
translate: 0.1
scale: 0.5
shear: 0.0
perspective: 0.0
flipud: 0.0
fliplr: 0.5
bgr: 0.0
mosaic: 1.0
mixup: 0.0
cutmix: 0.0
copy_paste: 0.0
copy_paste_mode: flip
auto_augment: randaugment
erasing: 0.4
cfg: null
tracker: botsort.yaml
save_dir: E:\pythonProject\ComputerVisionProject\YOLO\ultralytics-segment-GAN\my_project\runs\segment\mixed\train-mixed_1_1`

Dataset config:

path: E:\pythonProject\ComputerVisionProject\YOLO\ultralytics-segment-GAN\my_project\datasets\mixed_crack_dataset\1_1\dataset
train: .\images\train  # train images (relative to 'path') 128 images
val: .\images\val  # val images (relative to 'path') 128 images
test:  # test images (optional)

names:
  0: crack`

traning code
`import os
from ultralytics import YOLO

if __name__ == '__main__':   
    model = YOLO("yolo11m-seg.yaml")  # build a new model from YAML

    # Train the model
    results = model.train(data=r"my_project/datasets/mixed_crack_dataset/1_1/mixed_1_1.yaml", 
                          epochs=2000, imgsz=1024, device=0, batch=4, patience=0,
                          pretrained=False, amp=False, optimizer="AdamW")`

validation code
`import os
from ultralytics import YOLO

if __name__ == '__main__':   
    model = YOLO(r"my_project\runs\segment\traditional\train-traditional_1_1\weights\best.pt")

    # Train the model
    metrics = model.val(data=r"my_project/datasets/mixed_crack_dataset/1_1/mixed_1_1.yaml", 
                          epochs=2000, imgsz=1024, device=0, batch=4)
    print(f"map50-95: {metrics.box.map}")  # map50-95
    print(f"map50: {metrics.box.map50}")  # map50
    print(f"map75: {metrics.box.map75}")  # map75
    print(f"map50-95: {metrics.seg.map}")  # map50-95
    print(f"map50: {metrics.seg.map50}")  # map50
    print(f"map75: {metrics.seg.map75}")  # map75`

### Additional

the training results: Training results，it shows the segment metrics is normal, which is not 0 during the training prcess.

<img width="3600" height="1200" alt="Image" src="https://github.com/user-attachments/assets/e9e541fc-d348-46a3-b7f5-f5468309e0a9" />

But when I use the checkpoint best.pt to validate, it meets the issue of segment metrics. Bounding box metrics is normal, which is not 0. Segment metrics is abnormal, which is all 0. My validation code and results are as follows.The snapshot of validation results, it is the resutls of running the validation code above. 

<img width="2327" height="361" alt="Image" src="https://github.com/user-attachments/assets/3cf3e8a2-f64a-48fc-b597-b9a88b9d9d25" />

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @cjdfree, thanks for opening this issue and for using Ultralytics 🚀 This is an automated response to help get things moving quickly — an Ultralytics engineer will also assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) so we can reproduce the segmentation metrics behavior you’re seeing. Since your training metrics look normal but validation seg metrics are all zero, please include:
- Exact commands used for both train and val (CLI if possible) and corresponding full logs
- The exact model/weights file used for validation (e.g., path to best.pt) and how it was produced
- A tiny dataset subset (e.g., 5–10 images) with corresponding masks/labels that reproduces the issue
- Your data YAML and an example of label file structure for one val image
- A screenshot or copy of the printed metrics during training and during val
- Confirmation you’re using the same data.yaml and class names for train and val
- Any custom settings impacting segmentation (e.g., overlap_mask, mask_ratio, retina_masks) and whether changing them affects results

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Please upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a Python>=3.8 environment with PyTorch>=1.8 to verify the issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

After upgrading, re-run:
- Training: `yolo segment train model=yolo11m-seg.pt data=path/to/data.yaml imgsz=1024 batch=4 device=0`
- Validation: `yolo segment val model=path/to/best.pt data=path/to/data.yaml imgsz=1024 device=0`

and share the full console output if the problem persists.

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Google Cloud Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- Amazon Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- Docker Image. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thanks again for the detailed report and environment info 🙏 Please share the MRE and artifacts above so we can reproduce and help you faster.

**Comment by Y-T-G** [comment]
Can you provide the output after running this command in terminal: `yolo checks`?

**Comment by Y-T-G** [comment]
I can't reproduce this. Are you sure you're loading the same model?

**Comment by cjdfree** [comment]
Yes, it is the same model.

**Comment by Y-T-G** [comment]
Did the final evaluation during training show the correct results?

**Comment by cjdfree** [comment]
Yes, the result of the  final evaluation during training, the segment metrics are all 0, too

**Comment by cjdfree** [comment]
Another similar result also appears when training the object detection task. The metrics results exit, but it is abnormal when validation. 

<img width="1689" height="796" alt="Image" src="https://github.com/user-attachments/assets/7e42416f-c084-4a99-95d3-7ba9b088ed4a" />  

<img width="1407" height="271" alt="Image" src="https://github.com/user-attachments/assets/fcd0a9bc-c7ba-48aa-8a35-9979267df642" />

**Comment by Y-T-G** [comment]
You can try removing `amp=False`

**Comment by cjdfree** [comment]
This solution may not be the primary cause to this phenomena, whatever the value "amp" is, sometimes the metrics will be 0. Sometimes is normal.

**Comment by glenn-jocher** [comment]
Thanks for the added context—intermittent all‑zero Mask metrics usually mean the validator isn’t seeing polygon masks in the val split or it’s running with different settings than training. `model.val()` uses `rect=True` by default, so for an apples‑to‑apples check please run the same dataset with rectangular batching disabled and share the text output (not screenshots):
```bash
yolo val model="path\to\best.pt" data="path\to\mixed_1_1.yaml" imgsz=1024 batch=4 device=0 rect=False plots=True
```
Also confirm that train and val truly use the same `data.yaml` (your snippets show `mixed_6_1.yaml` vs `mixed_1_1.yaml`), and that a sample `labels/val/*.txt` contains class id followed by polygon x y pairs (not just 4 box numbers). After any label fixes, delete the `*.cache` files inside both `labels/train` and `labels/val`. For more detail, you can print the per‑class mask/box summary:
```python
metrics = model.val(data="path/to/mixed_1_1.yaml", imgsz=1024, device=0, batch=4, rect=False, plots=True)
print(metrics.summary(decimals=4))
```
If the issue persists, please paste the full `yolo checks` text output and the last epoch’s validation log so we can verify the exact validator path and args. References: the validator that computes both Box and Mask metrics is documented in the SegmentationValidator reference, and `model.val()` defaults (including `rect=True`) are shown in the Model.val API; segment metrics accessors and `summary()` are in SegmentMetrics. See [SegmentationValidator reference](https://docs.ultralytics.com/reference/models/yolo/segment/val/), [Model Validation guide](https://docs.ultralytics.com/modes/val/), [Model.val API](https://docs.ultralytics.com/reference/engine/model/#val), and [SegmentMetrics.summary](https://docs.ultralytics.com/reference/utils/metrics/#summary).
